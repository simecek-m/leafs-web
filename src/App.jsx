import { Auth0Provider } from "@auth0/auth0-react";
import Home from "page/Home";
import { BrowserRouter, Route, Routes } from "react-router-dom";

const App = () => {
  return (
    <Auth0Provider
      audience="todo-server"
      domain={process.env.REACT_APP_AUTH_DOMAIN}
      clientId={process.env.REACT_APP_AUTH_CLIENT_ID}
      redirectUri={window.location.origin}
    >
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home />} />
        </Routes>
      </BrowserRouter>
    </Auth0Provider>
  );
};

export default App;
