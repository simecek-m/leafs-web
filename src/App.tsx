import { Auth0Provider } from "@auth0/auth0-react";
import { HOME_PATH, INDEX_PATH, REGISTRATION_PATH } from "constant";
import Dashboard from "page/Dashboard";
import Home from "page/Home";
import ProtectedRoute from "page/ProtectedRoute";
import Registration from "page/Registration";
import { FC } from "react";
import { BrowserRouter, Route, Routes } from "react-router-dom";

const App: FC = () => {
  return (
    <Auth0Provider
      domain={process.env.REACT_APP_AUTH_DOMAIN}
      clientId={process.env.REACT_APP_AUTH_CLIENT_ID}
      authorizationParams={{
        audience: process.env.REACT_APP_AUTH_AUDIENCE,
        redirect_uri: window.location.origin,
      }}
    >
      <BrowserRouter>
        <Routes>
          <Route path={HOME_PATH} element={<Home />} />
          <Route path={INDEX_PATH} element={<ProtectedRoute />}>
            <Route index element={<Dashboard />} />
            <Route path={REGISTRATION_PATH} element={<Registration />} />
          </Route>
        </Routes>
      </BrowserRouter>
    </Auth0Provider>
  );
};

export default App;
