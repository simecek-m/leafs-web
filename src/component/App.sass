:root
  --transparent: rgb(0 0 0 / 0%)
  --brand: rgb(111 234 101)
  --on-brand: black
  --dark-gray: rgb(21 21 21)
  --light-gray: rgb(243 243 243)
  --topbar-height: 40px

body
  margin: 0
  font-family: "Segoe UI", sans-serif

.theme-dark
  --background: black
  --color: white
  --surface: var(--dark-gray)
  --on-surface: white
  --widget-background: var(--dark-gray)

.theme-light
  --background: white
  --color: black
  --surface: var(--light-gray)
  --on-surface: black
  --widget-background: white
