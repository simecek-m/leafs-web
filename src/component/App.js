import React from "react";
import "animate.css";
import { HomePage } from "./page/Home";
import { connect } from "react-redux";
import "component/App.sass";

function App({ theme }) {
  return (
    <div className={`theme-${theme}`}>
      <HomePage />
    </div>
  );
}

const mapStateToProps = state => {
  return {
    theme: state.theme
  };
};
export default connect(mapStateToProps)(App);
