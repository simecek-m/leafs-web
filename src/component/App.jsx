import React from "react";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import HomePage from "page/HomePage";
import { useTheme } from "provider/theme";
import { Toaster } from "react-hot-toast";
import Page from "page/Page";
import ListPage from "page/ListPage";
import AnonymousRoute from "router/AnonymousRoute";
import ProtectedRoute from "router/ProtectedRoute";
import NewListPage from "page/NewListPage";
import ListDetailPage from "page/ListDetailPage";
import NoneListSelectedPage from "page/NoneListSelectedPage";
import "component/App.sass";

export default function App() {
  const { theme } = useTheme();
  return (
    <div className={`theme-${theme}`}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Page />}>
            <Route element={<AnonymousRoute />}>
              <Route index element={<HomePage />} />
            </Route>
            <Route path="list" element={<ProtectedRoute />}>
              <Route element={<ListPage />}>
                <Route index element={<NoneListSelectedPage />} />
                <Route path="new" element={<NewListPage />} />
                <Route path=":id" element={<ListDetailPage />} />
              </Route>
            </Route>
          </Route>
        </Routes>
      </BrowserRouter>
      <Toaster position="bottom-right" reverseOrder={false} />
    </div>
  );
}
