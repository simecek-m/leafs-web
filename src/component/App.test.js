import React from "react";
import App from "component/App";
import { shallow } from "enzyme";

describe("App component", () => {
  test("should render App", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
