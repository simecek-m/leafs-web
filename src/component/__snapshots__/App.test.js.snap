// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Connect(AnonymousRoute)
      component={[Function]}
      path="/login"
    />
    <Connect(ProtectedRoute)
      component={[Function]}
      exact={true}
      path="/"
    />
    <Connect(ProtectedRoute)
      component={[Function]}
      path="/profile"
    />
    <Connect(ProtectedRoute)
      component={[Function]}
      path="/todos/:id"
    />
    <Connect(ProtectedRoute)
      component={[Function]}
      path="/about"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</Router>
`;
